### PR TITLE
feat (aws/ecs): Surface task availability zone in ECS instance details

### PR DESCRIPTION
**Pull request by wlbrough**
This update adds the Task availability zone to the ECS instance details. This is based on an outstanding feature request ([#6116](https://github.com/spinnaker/spinnaker/issues/6116)). Test assertions have been added to support the changes.

If this pull request is merged, I will submit a separate PR for deck to complete the feature request.

### CODE REVIEW

File: clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/view/EcsInstanceProvider.java [hunk 1]
```diff
@@ -72,7 +72,6 @@ public EcsTask getInstance(String account, String region, String id) {
     List<Map<String, Object>> healthStatus =
         containerInformationService.getHealthStatus(id, serviceName, account, region);
     String address = containerInformationService.getTaskPrivateAddress(account, region, task);
-    String zone = containerInformationService.getTaskZone(account, region, task);
```
Comment (mattgogerly) [inline clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/view/EcsInstanceProvider.java:75]: Should this be the default if availabilityZone is not specified?

Comment (wlbrough) [reply clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/view/EcsInstanceProvider.java:75]: That makes sense here. What are your thoughts on implementing that like this (similar to `networkInterface` immediately below)?

```java
String zone =
    task.getAvailabilityZone() != null
        && !task.getAvailabilityZone().isEmpty()
    ? task.getAvailabilityZone()
    : containerInformationService.getTaskZone(account, region, task);
```

Comment (wlbrough) [reply clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/view/EcsInstanceProvider.java:75]: Updated!

**Review by deverton — APPROVED** [review]
Looks good to me.

**Review by mattgogerly — APPROVED** [review]

**Pull request merged**